File: src/apis/process.js
import { invoke } from "@tauri-apps/api/core"

export async function process_run_apps(paths, login) {
     return await invoke("process_run_apps",{paths,login})
}
export async function process_run_app(file) {
    return await invoke("process_run_app",{file})
}

export async function process_close_apps(files) {
    return await invoke("process_close_apps",{files})
}

export async function process_close_app(fileName,delay) {
    return await invoke("process_close_app",{fileName,delay})
}

File: src-tauri/src/apis/process.rs
use crate::errors::Result;
use services::process;
use winsys::process::pid::Pid;

#[tauri::command(async)]
pub fn process_run_apps(paths: Vec<String>, login: Option<String>) -> Result<()> {
    Ok(process::process_run_apps(&paths,&login)?)
}

#[tauri::command(async)]
pub fn process_run_app(file: String) -> Result<Vec<Pid>> {
    Ok(process::process_run_app(file.as_str())?)
}

#[tauri::command(async)]
pub fn process_close_apps(files: Vec<String>) -> Result<()> {
    Ok(process::process_close_apps(&files)?)
}

#[tauri::command(async)]
pub fn process_close_app(file_name: String,delay:u64) -> Result<()> {
    Ok(process::process_close_app(file_name.as_str(),delay)?)
}
